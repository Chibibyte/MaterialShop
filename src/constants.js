export default {
    BUTTONICONS: {
        AUTHENTICATION: 'mdi-login-variant',
        ADDRESS: 'mdi-package-variant',
        BANKACCOUNT: 'mdi-bank',
        SHIPPING: 'mdi-truck',
        PAYMENT: 'mdi-cash',
        SUMMARY: 'mdi-text-box-check-outline',
        CHECKOUT_COMPLETE: 'mdi-check',
        ADD: 'mdi-plus',
        REMOVE: 'mdi-minus',
        CANCEL: 'mdi-window-close',
        CONFIRM: 'mdi-check',
        SHOPPING: 'mdi-shopping',
        SUCCESS: 'mdi-check',
        FAILURE: 'mdi-window-close',
        WARNING: 'mdi-alert-circle-outline'
    }
}