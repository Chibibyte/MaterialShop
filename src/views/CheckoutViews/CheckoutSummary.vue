<template>
  <CheckoutBase
    :next_disabled="!checkoutStates.summary"
    title="Summary"
    subtitle="Please verify that your order is correct"
    next_target="complete"
    :next_icon="$constants.BUTTONICONS.CHECKOUT_COMPLETE"
    :icon="$constants.BUTTONICONS.SUMMARY"
    invisible
  >
    <v-row>
      <v-col>
        <h1>Order</h1>
        <ItemListViewItem
          v-for="item in checkout.order"
          :key="item.id"
          :item="{ title: materialNames[item.id], price: '2.99' }"
          :initAmount="item.amount"
          hideButtons
          flexRow
          class="mb-5"
          readonly
        />
        <h1>DeliveryAddress</h1>
        <Address
          class="mb-5"
          readonly
          dense
          :value="checkout.deliveryAddress"
        />
        <h1>Bankaccount</h1>
        <Bankaccount
          class="mb-5"
          readonly
          dense
          :value="checkout.bankaccount"
        />
        <h1>Shipping</h1>
        <Shipping class="mb-5" readonly dense :value="checkout.shipping" />
        <h1>Payment</h1>
        <Payment class="mb-5" readonly dense :value="checkout.payment" />
      </v-col>
      <v-col>
        <h1>Price</h1>
        <SummaryCost
          class="mb-5"
          readonly
          dense
          :value="{
            itemsCost: itemsCost + '€',
            shippingCost: shippingCost + '€',
            totalCost: totalCost + '€',
          }"
        />
        <NavButton
          routerPush
          :runBefore="completeOrder"
          to="complete"
          :icon="$constants.BUTTONICONS.CHECKOUT_COMPLETE"
          name="Order now"
        />
      </v-col>
    </v-row>
  </CheckoutBase>
</template>

<script>
import CheckoutBase from "@/views/CheckoutViews/CheckoutBase.vue";
import Address from "@/components/Address.vue";
import Shipping from "@/components/Shipping.vue";
import Payment from "@/components/Payment.vue";
import Bankaccount from "@/components/Bankaccount.vue";
import ItemListViewItem from "@/components/ItemListViewItem.vue";
import SummaryCost from "@/components/SummaryCost.vue";
import NavButton from "@/components/NavButton.vue";
export default {
  name: "CheckoutSummary",
  components: {
    CheckoutBase,
    Address,
    Shipping,
    Payment,
    Bankaccount,
    ItemListViewItem,
    SummaryCost,
    NavButton,
  },
  data() {
    return {
      checkoutStates: this.$store.state.checkout.checkoutStates,
      checkout: this.$store.state.checkout,
      noncheckout: this.$store.state.noncheckout,
      materialNames: this.$store.state.noncheckout.materialNames,
      shippingCost: this.$store.state.checkout.shipping.price,
      itemsCost:
        Math.round(
          this.$store.state.checkout.order.reduce((acc, cur) => {
            return acc + cur.amount * 2.99;
          }, 0) * 100
        ) / 100,
    };
  },
  methods: {
    completeOrder() {
      this.$store.commit("checkout/clearCheckoutStates");
      this.$store.commit("checkout/clearCheckoutContent");
      this.$store.commit("noncheckout/clearCart");
      this.$store.commit("checkout/setCheckoutState", {
        chState: "summary",
        value: true,
      });
    },
  },
  computed: {
    console: () => console,
    totalCost() {
      return (
        Math.round((Number(this.itemsCost) + Number(this.shippingCost)) * 100) /
        100
      );
    },
  },
};
</script>