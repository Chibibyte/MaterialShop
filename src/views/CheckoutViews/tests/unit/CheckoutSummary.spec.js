
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CheckoutSummary from '@/views/CheckoutViews/CheckoutSummary.vue';
import constants from '@/constants.js'

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.prototype.$constants = constants;

let store;
let state;
let modules;
let stubs = [
    'v-container',
    'v-row',
    'v-col'
]
let wrapper;

beforeEach(() => {
    state = {}
    modules = {
        noncheckout: {
            namespaced: true,
            state() {
                return {
                    cart: [
                        {
                            id: 3,
                            amount: 4
                        },
                        {
                            id: 4,
                            amount: 2
                        }
                    ],
                    materialNames: [
                        'alpha',
                        'amount',
                        'anna',
                        'anno',
                        'bread',
                        'tennis',
                        'toast'
                    ]
                }
            },
            mutations: {
                clearCart: jest.fn()
            }
        },
        checkout: {
            namespaced: true,
            state() {
                return {
                    checkoutStates: {
                        authentication: false,
                        deliveryAddress: false,
                        bankaccount: false,
                        shipping: false,
                        payment: false,
                        summary: false,
                    },
                    shipping: {
                        price: 2.99
                    },
                    order: []
                }
            },
            mutations: {
                clearCheckoutStates: jest.fn(),
                clearCheckoutContent: jest.fn()
            }
        }
    }
    store = new Vuex.Store({
        state,
        modules
    })
    wrapper = shallowMount(CheckoutSummary, { stubs, store, localVue });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
});

describe('set output props', () => {
    test('all props set', async () => {
        await wrapper.setProps({
            icon: 'test icon', title: 'test title', subtitle: 'test subtitle',
            invisible: true, next_disabled: true, next_target: 'test next_target', next_icon: 'test next_icon', next_name: 'test next_name'
        })
        expect(wrapper.element).toMatchSnapshot();
    })

    test('all props except invisible', async () => {
        await wrapper.setProps({
            icon: 'test icon', title: 'test title', subtitle: 'test subtitle',
            next_disabled: true, next_target: 'test next_target', next_icon: 'test next_icon', next_name: 'test next_name'
        })
        expect(wrapper.element).toMatchSnapshot();
    })
})

describe('methods', () => {
    test('completeOrder', () => {
        wrapper.vm.completeOrder();
        expect(modules.checkout.mutations.clearCheckoutStates.mock.calls).toHaveLength(1);
        expect(modules.checkout.mutations.clearCheckoutContent.mock.calls).toHaveLength(1);
        expect(modules.noncheckout.mutations.clearCart.mock.calls).toHaveLength(1);
    })
})