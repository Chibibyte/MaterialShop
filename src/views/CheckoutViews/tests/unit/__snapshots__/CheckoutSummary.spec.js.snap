// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<checkoutbase-stub
  icon="mdi-text-box-check-outline"
  invisible="true"
  next_disabled="true"
  next_icon="mdi-check"
  next_name="next"
  next_target="complete"
  subtitle="Please verify that your order is correct"
  title="Summary"
>
  <v-row-stub>
    <v-col-stub>
      <h1>
        Order
      </h1>
        
      <h1>
        DeliveryAddress
      </h1>
       
      <address-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
       
      <h1>
        Bankaccount
      </h1>
       
      <bankaccount-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
       
      <h1>
        Shipping
      </h1>
       
      <shipping-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
        value="[object Object]"
      />
       
      <h1>
        Payment
      </h1>
       
      <payment-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
    </v-col-stub>
     
    <v-col-stub>
      <h1>
        Price
      </h1>
       
      <summarycost-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
        value="[object Object]"
      />
       
      <navbutton-stub
        icon="mdi-check"
        name="Order now"
        routerpush="true"
        runbefore="function () { [native code] }"
        to="complete"
      />
    </v-col-stub>
  </v-row-stub>
</checkoutbase-stub>
`;

exports[`set output props all props except invisible 1`] = `
<checkoutbase-stub
  icon="mdi-text-box-check-outline"
  invisible="true"
  next_disabled="true"
  next_icon="mdi-check"
  next_name="next"
  next_target="complete"
  subtitle="Please verify that your order is correct"
  title="Summary"
>
  <v-row-stub>
    <v-col-stub>
      <h1>
        Order
      </h1>
        
      <h1>
        DeliveryAddress
      </h1>
       
      <address-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
       
      <h1>
        Bankaccount
      </h1>
       
      <bankaccount-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
       
      <h1>
        Shipping
      </h1>
       
      <shipping-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
        value="[object Object]"
      />
       
      <h1>
        Payment
      </h1>
       
      <payment-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
    </v-col-stub>
     
    <v-col-stub>
      <h1>
        Price
      </h1>
       
      <summarycost-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
        value="[object Object]"
      />
       
      <navbutton-stub
        icon="mdi-check"
        name="Order now"
        routerpush="true"
        runbefore="function () { [native code] }"
        to="complete"
      />
    </v-col-stub>
  </v-row-stub>
</checkoutbase-stub>
`;

exports[`set output props all props set 1`] = `
<checkoutbase-stub
  icon="mdi-text-box-check-outline"
  invisible="true"
  next_disabled="true"
  next_icon="mdi-check"
  next_name="next"
  next_target="complete"
  subtitle="Please verify that your order is correct"
  title="Summary"
>
  <v-row-stub>
    <v-col-stub>
      <h1>
        Order
      </h1>
        
      <h1>
        DeliveryAddress
      </h1>
       
      <address-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
       
      <h1>
        Bankaccount
      </h1>
       
      <bankaccount-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
       
      <h1>
        Shipping
      </h1>
       
      <shipping-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
        value="[object Object]"
      />
       
      <h1>
        Payment
      </h1>
       
      <payment-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
      />
    </v-col-stub>
     
    <v-col-stub>
      <h1>
        Price
      </h1>
       
      <summarycost-stub
        class="mb-5"
        dense="true"
        readonly="true"
        validate="default() {}"
        value="[object Object]"
      />
       
      <navbutton-stub
        icon="mdi-check"
        name="Order now"
        routerpush="true"
        runbefore="function () { [native code] }"
        to="complete"
      />
    </v-col-stub>
  </v-row-stub>
</checkoutbase-stub>
`;
